blockchain/chain_sync/src/sync.rs

```diff
@@ -585,23 +584,32 @@ where
         // check messages to ensure valid state transitions
         self.check_blk_msgs(block.clone(), &parent_tipset)?;
 
+        // TODO use state_root instead of parent_tipset.parent_state()
```

Comment:
can you explain this?

Reply:
Rather than using the parent state root, use the computed state root returned (when its implemented) [here](https://github.com/filecoin-project/lotus/blob/testnet/3/chain/stmgr/stmgr.go#L63:25).